src/traces/scattermapbox/plot.js

```diff
@@ -42,11 +50,21 @@ function ScatterMapbox(subplot, uid) {
 
 var proto = ScatterMapbox.prototype;
 
-proto.addSource = function(k, opts) {
-    this.subplot.map.addSource(this.sourceIds[k], {
-        type: 'geojson',
-        data: opts.geojson
-    });
+proto.addSource = function(k, opts, cluster) {
+    var sourceOpts = {
+            type: 'geojson',
+            data: opts.geojson,
+    };
+    
+    if(cluster.enabled) {
+        Lib.extendFlat({}, sourceOpts, {
```

Comment:
My apologies. Please remove the first argument to this function.
This line should be
```js
Lib.extendFlat(sourceOpts, {
```